Fix composer error by adding in a task name for the actual running of…

seamuslee001 opened the pull request:
… the Compile task event

@totten @demeritcowboy I think this will resolve #19

seamuslee001 committed:
Fix composer error by adding in a task name for the actual running of the Compile task event

demeritcowboy commented:
Thanks. It does seem to fix the compile step but then I get the same error as the test run here.

```
In Command.php line 301:

  [TypeError]
  Return value of "Civi\CompilePlugin\Command\CompileCommand::execute()" must be of the type int, "null" returned.


Exception trace:
  at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Command/Command.php:301
 Symfony\Component\Console\Command\Command->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:1015
 Symfony\Component\Console\Application->doRunCommand() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:334
 Composer\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:130
 Composer\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:83
 require() at C:\ProgramData\ComposerSetup\bin\composer.phar:29
```

totten committed:
Commands - Always return an exit code

totten committed:
MultiStepTaskTest - Add smaller assertion for `@export`

The other assertions combine `@export` with some data look-ups.

This just does a basic export.

totten commented:
Sounds like the newer version of `composer` also has a stricter expectation for return-values of each `FooCommand::execute()` method. That seems simple enough...

But on my local, with php74 and latest composer build (`composer self-update --snapshot`), it's still not passing. There seems to be something affecting the `@export` directives here:

https://github.com/civicrm/composer-compile-plugin/blob/v0.17/tests/MultiStepTaskTest.php#L43-L47

totten committed:
Update putenv calls for compatibility wiht 2.3+

Aside: The `EnvHelper::set()` turns out to be basically identical to `Platform::putenv()`, so
I also simplified/swapped the calls to that.

totten commented:
Test matrix passes now.
